Better error message if value is missing on the CLI for an SPI

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

dist/quarkus

### Describe the bug

When starting Keycloak with a missing value for an SPI

```
bin/kc.sh start --spi-events-listener-jboss-logging-success-level
```

the error message `NoSuchElementException` and a stacktrace appears. 



### Version

main

### Expected behavior

Users should get advice that they missed the value for the SPI parameter.



### Actual behavior

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: java.util.NoSuchElementException
        at java.base/java.util.ArrayList$Itr.next(ArrayList.java:970)
        at org.keycloak.quarkus.runtime.cli.Picocli.parseArgs(Picocli.java:522)
        at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:66)
        ... 6 more
```


### How to Reproduce?

Run 

```
bin/kc.sh start --spi-events-listener-jboss-logging-success-level
```

### Anything else?

As this is very frustrating for first-time users and once wouldn't find the cause without looking at the Java code, this should be fixed soon.

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakMain.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakMain.java
@@ -41,7 +41,7 @@ import org.keycloak.Config;
 import org.keycloak.models.KeycloakSessionFactory;
 import org.keycloak.models.utils.KeycloakModelUtils;
 import org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler;
-import org.keycloak.quarkus.runtime.cli.NonCliPropertyException;
+import org.keycloak.quarkus.runtime.cli.PropertyException;
 import org.keycloak.quarkus.runtime.cli.Picocli;
 import org.keycloak.common.Version;
 import org.keycloak.quarkus.runtime.cli.command.Start;
@@ -64,7 +64,16 @@ public class KeycloakMain implements QuarkusApplication {
 
     public static void main(String[] args) {
         System.setProperty("kc.version", Version.VERSION);
-        List<String> cliArgs = Picocli.parseArgs(args);
+        List<String> cliArgs = null;
+        try {
+            cliArgs = Picocli.parseArgs(args);
+        } catch (PropertyException e) {
+            ExecutionExceptionHandler errorHandler = new ExecutionExceptionHandler();
+            PrintWriter errStream = new PrintWriter(System.err, true);
+            errorHandler.error(errStream, e.getMessage(), null);
+            System.exit(ExitCode.USAGE);
+            return;
+        }
 
         if (cliArgs.isEmpty()) {
             cliArgs = new ArrayList<>(cliArgs);
@@ -83,7 +92,7 @@ public class KeycloakMain implements QuarkusApplication {
 
             try {
                 Picocli.validateNonCliConfig(cliArgs, new Start(), new PrintWriter(System.out, true));
-            } catch (NonCliPropertyException e) {
+            } catch (PropertyException e) {
                 errorHandler.error(errStream, e.getMessage(), null);
                 System.exit(ExitCode.USAGE);
                 return;

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/ExecutionExceptionHandler.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/ExecutionExceptionHandler.java
@@ -42,7 +42,7 @@ public final class ExecutionExceptionHandler implements CommandLine.IExecutionEx
 
     @Override
     public int handleExecutionException(Exception cause, CommandLine cmd, ParseResult parseResult) {
-        if (cause instanceof NonCliPropertyException) {
+        if (cause instanceof PropertyException) {
             PrintWriter writer = cmd.getErr();
             writer.println(cmd.getColorScheme().errorText(cause.getMessage()));
             return ShortErrorMessageHandler.getInvalidInputExitCode(cause, cmd);

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
@@ -297,7 +297,7 @@ public final class Picocli {
                 }
 
                 if (!PropertyMapperParameterConsumer.isExpectedValue(mapper.getExpectedValues(), value)) {
-                    throw new NonCliPropertyException(PropertyMapperParameterConsumer.getErrorMessage(mapper.getFrom(),
+                    throw new PropertyException(PropertyMapperParameterConsumer.getErrorMessage(mapper.getFrom(),
                             value, mapper.getExpectedValues(), mapper.getExpectedValues()) + ". From ConfigSource " + configSource.getName());
                 }
             }
@@ -595,7 +595,7 @@ public final class Picocli {
         cmd.getOut().println(message);
     }
 
-    public static List<String> parseArgs(String[] rawArgs) {
+    public static List<String> parseArgs(String[] rawArgs) throws PropertyException {
         if (rawArgs.length == 0) {
             return List.of();
         }
@@ -615,6 +615,12 @@ public final class Picocli {
                 iterator.remove();
 
                 if (!arg.contains(ARG_KEY_VALUE_SEPARATOR)) {
+                    if (!iterator.hasNext()) {
+                        if (arg.startsWith("--spi")) {
+                            throw new PropertyException(String.format("spi argument %s requires a value.", arg));
+                        }
+                        return args;
+                    }
                     String next = iterator.next();
 
                     if (!next.startsWith("--")) {

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/PropertyException.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/PropertyException.java
@@ -17,9 +17,9 @@
 
 package org.keycloak.quarkus.runtime.cli;
 
-public class NonCliPropertyException extends RuntimeException {
+public class PropertyException extends RuntimeException {
 
-    public NonCliPropertyException(String message) {
+    public PropertyException(String message) {
         super(message);
     }
 

